agent/serf.go

```diff
@@ -131,3 +148,49 @@ type retryingClient struct {
 	client *serf.RPCClient
 	config serf.Config
 }
+
+// MockSerfClient is a mock/fake Serf Client used in testing
+type MockSerfClient struct {
+	members []serf.Member
+	coords  map[string]*coordinate.Coordinate
+}
+
+// NewMockSerfClient is an helper function used to create a mock/fake Serf Client
+// used in testing
+func NewMockSerfClient(members []serf.Member, coords map[string]*coordinate.Coordinate) (client *MockSerfClient, err error) {
+	return &MockSerfClient{
+		members: members,
+		coords:  coords,
+	}, nil
+}
+
+// Members is a function that return the (fake) Serf member nodes
```

Comment:
that return**s** ... You can remove `fake` (as it will be obvious from the test context) when you move this implementation to serf_test.go